metrics/web/lib/common/presentation/injector/widget/injection_container.dart

```diff
@@ -173,6 +201,30 @@ class _InjectionContainerState extends State<InjectionContainer> {
     );
   }
 
+  /// A closure to be called when the [AuthNotifier] changes.
+  void _authNotifierListener() {
+    final updatedUserProfile = _authNotifier.userProfileModel;
+
+    _themeNotifier.changeTheme(updatedUserProfile?.selectedTheme);
+  }
+
+  /// A closure to be called when the [ThemeNotifier] changes.
+  void _themeNotifierListener() {
+    final selectedTheme = _themeNotifier.selectedTheme;
+
+    _authNotifier.changeTheme(selectedTheme);
+
+    final currentUserProfile = _authNotifier.userProfileModel;
+
+    if (currentUserProfile == null) {
+      return;
+    }
+
+    _authNotifier.updateUserProfile(
+      currentUserProfile.copyWith(selectedTheme: selectedTheme),
```

Comment:
The same comment as above. We should pass the `UserProfileModel` with only changed field to the `AuthNotifier` and then merge this user profile with the current one inside the `AuthNotifier`.